### PR TITLE
add shared subscription priority level

### PR DESCRIPTION
**Pull request by fourhu**
<--
### Contribution Checklist
  
  - Name the pull request in the form "[Issue XYZ][component] Title of the pull request", where *XYZ* should be replaced by the actual issue number.
    Skip *Issue XYZ* if there is no associated github issue for this pull request.
    Skip *component* if you are unsure about which is the best component. E.g. `[docs] Fix typo in produce method`.

  - Fill out the template below to describe the changes contributed by the pull request. That will give reviewers the context they need to do the review.
  
  - Each pull request should address only one issue, not mix up code from multiple issues.
  
  - Each commit in the pull request has a meaningful commit message

  - Once all items of the checklist are addressed, remove the above text and this checklist, leaving only the filled out template below.

**(The sections below can be removed for hotfixes of typos)**
-->

*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: [#134](https://github.com/apache/pulsar/issues/134)

### Motivation
support priority level for subscription of shared subscription type

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*
consumer options support priority level for subscription of shared subscription type

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): no
  - The public API: no
  - The schema: no
  - The default values of configurations: yes
  - The wire protocol: no

### Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / GoDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: pulsar/consumer_impl.go [hunk 1]
```diff
@@ -72,6 +73,14 @@ func newConsumer(client *client, options ConsumerOptions) (Consumer, error) {
 		return nil, newError(SubscriptionNotFound, "subscription name is required for consumer")
 	}
 
+	if options.PriorityLevel <= 0 {
+		options.PriorityLevel = 0
```
Comment (liangyuanpeng) [inline pulsar/consumer_impl.go:77]: It should be return error. Ref: https://github.com/apache/pulsar/blob/master/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBuilderImpl.java#L254

File: pulsar/consumer_impl.go [hunk 2]
```diff
@@ -72,6 +73,14 @@ func newConsumer(client *client, options ConsumerOptions) (Consumer, error) {
...
+	}
+
+	if options.Type != Shared {
```
Comment (liangyuanpeng) [inline pulsar/consumer_impl.go:80]: Does we need this?

**Review by liangyuanpeng — COMMENTED** [review]
Thanks for your work and added some comment.

File: pulsar/consumer_impl.go (hunk 1 above)
Comment (fourhu) [reply pulsar/consumer_impl.go:77]: ok, i will add then error later

File: pulsar/consumer_impl.go (hunk 2 above)
Comment (fourhu) [reply pulsar/consumer_impl.go:80]: Is PriorityLevel effectived when SubscriptionType is Shared?

**Review by merlimat — COMMENTED** [review]
We should have a unit test to check the priorities are applied correctly

**Comment by fourhu** [comment]
> or



> 

ok, I will add unit test later.

**Pull request closed without merging**